### DISCUSSION TITLE
Parsing a formula using C++/Python API

### DISCUSSION THREAD

**Discussion by pkalita595** [cvc5/cvc5#9111] [Q&A] [answered]
Is there a way to parse a formula from a file and get its relevant object using C++/Python API?

**Comment by kfriedberger** [comment, marked as answer]
Related topic: https://github.com/cvc5/cvc5/discussions/8345: **Is there a way of using APIs (Java) to read in an smt2 file ?**
Answer: **No, not yet**

**Reply by alex-ozdemir** [reply]
Revisiting this. The API has been extended to allow this, as in [this example](https://cvc5.github.io/docs-ci/docs-main/examples/parser.html).

**Reply by Flower-Wang1024** [reply]
Hello! This example just use the function check_sat(), if I want to get model of smt2 file, what should I do? Because there is no declared terms or sorts.

**Reply by alex-ozdemir** [reply]
All the standard API functions are still available, including those for getting a model/getting the value of a term within a model.
You might also be interested in [this function](https://cvc5.github.io/docs-ci/docs-main/api/cpp/inputparser.html#_CPPv4N4cvc56parser11InputParser8nextTermEv) for parsing terms.
